blocks/main_loop.py

```diff
@@ -151,12 +149,7 @@ def run(self):
         # reset `profile.current`. Otherwise, it simply does not hurt.
         self.profile.current = []
 
-        # Sanity check for the most common case
-        if (self._model and isinstance(self._model, Model) and
```

Comment:
I think that it would be better to move this sanity check to `Model` and not to `GradientDescent`. Algorithms are the cleanest part of Blocks and I would like to avoid code bloat there.